flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedBackendSerializationProxy.java

```diff
@@ -44,7 +47,17 @@
  */
 public class KeyedBackendSerializationProxy<K> extends VersionedIOReadableWritable {
 
-	public static final int VERSION = 5;
+	public static final int VERSION = 6;
+
+	private static final Map<Integer, Integer> META_INFO_SNAPSHOT_FORMAT_VERSION_MAPPER = new HashMap<>();
+	static {
```

Comment:
Theoretically, no, but I think practice it would be best to keep track of this.
It just happens to be that the versions match for this case. See for example, `OperatorBackendSerializationProxy` where the versions don't match.

Reply:
Agree we should keep track of the mapping. I just felt we could improve the clarity of the tracking even a bit more, but I could not think of a better solution right now. But as I said I think we could try make it a bit more straightforward in some follow-up.

Just to sum it up. I think it is absolutely ok for this PR, but I have a feeling we could improve readability of this mapping in the future.